WcaOnRails/app/webpacker/components/EditSchedule/EditVenues/VenuePanel.js

```diff
@@ -134,10 +146,13 @@ function VenuePanel({
         </Card.Header>
         <Card.Description>
           <Card.Group itemsPerRow={2}>
-            {venue.rooms.map((room) => (
+            {venue.rooms.map((room, index) => (
               <RoomPanel
                 key={room.id}
                 room={room}
+                sendToFront={
+                  index > 0 ? () => dispatch(reorderRoom(venue.id, index, 0)) : undefined
```

Comment:
Why are you passing in the whole action (function) as a prop?
The `RoomPanel` component is able to access the dispatch on its own. I think passing the information which index this is should be enough, so that the panel can figure out the logic on its own.

Reply:
My reasoning was that a room should only directly dispatch actions which affect only that room. This action affects other rooms too, so is a venue-level action.

I don't mind changing it though. That would require passing in the venue id too, in addition to the index (though I could rewrite the reducer to take the room id instead of the venue id, and have it look up which venue the room is in, but that complicates the reducer). What do you think?